File: Practica_4/SCFunciones/FiguraGeometrica.java
package Practica_4.SCFunciones;

public class FiguraGeometrica {
    
    // círculo
    public double area(double radio) {
        return Math.PI * radio * radio;
    }

    // rectángulo
    public double area(double largo, double ancho) {
        return largo * ancho;
    }

    // triángulo rectángulo
    public double area(double base, float altura) {
        return (base * altura) / 2;
    }

    // trapecio
    public double area(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2;
    }

    // pentágono
    public double area(int lado) {
        return (1.0 / 4.0) * Math.sqrt(5 * (5 + 2 * Math.sqrt(5))) * lado * lado;
    }

    public static void main(String[] args) {
        FiguraGeometrica Fg = new FiguraGeometrica();

        System.out.println("círculo: " + Fg.area(5));
        System.out.println("rectángulo: " + Fg.area(4, 6));
        System.out.println("triángulo rectángulo: " + Fg.area(3, 4));
        System.out.println("trapecio: " + Fg.area(4, 6, 5));
        System.out.println("pentágono: " + Fg.area(3));
    }
}

File: Practica_4/SCFunciones/FiguraGeometrica.py
import math
#no podemos hacer sobre carga de funciones en python pero,
#podemos usar condicionales para que solo usemos un metodo
class FiguraGeometrica:
    
    def area(self, *args):
        if len(args) == 1:
            # círculo
            radio = args[0]
            return math.pi * radio * radio
        elif len(args) == 2:
            # rectángulo
            largo, ancho = args
            return largo * ancho
        elif len(args) == 3:
            # triángulo rectángulo o trapecio
            if isinstance(args[2], bool):
                # triángulo rectángulo
                base, altura, _ = args
                return (base * altura) / 2
            else:
                # trapecio
                baseMayor, baseMenor, altura = args
                return ((baseMayor + baseMenor) * altura) / 2
        elif len(args) == 2:
            # pentágono
            lado, apotema = args
            return (5 * lado * apotema) / 2
        else:
            return 0


if __name__ == "__main__":
    Fg = FiguraGeometrica()

    print("círculo:", Fg.area(5))
    print("rectángulo:", Fg.area(4, 6))
    print("triángulo rectángulo:", Fg.area(3, 4, True))
    print("trapecio:", Fg.area(4, 6, 5))
    print("pentágono:", Fg.area(3, 4))
    
#podemos simularla utilizando un solo método con parámetros 
# opcionales y lógica condicional.